plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -1678,7 +1700,7 @@ private void flushBuffer(boolean finished)
             }
 
             // The multipart upload API only accept the last part to be less than 5MB
-            if (bufferSize == buffer.length || (finished && bufferSize > 0)) {
+            if (bufferSize == MINIMAL_TARGET_PART_SIZE || (finished && bufferSize > 0)) {
```

Comment:
`bufferSize` should be able to exceed `MINIMAL_TARGET_PART_SIZE`.

It should really by:
```
if (bufferSize == partSize || (finished && bufferSize > 0)) {
```

Reply:
I wanted it to be `if (bufferSize == targetPartSize || (finished && bufferSize > 0)) {` but it doesn't matter now